### ISSUE TITLE
SponsorBlock - Auto-skip as an option

### ISSUE THREAD

**Issue by ScriptForAll** [FreeTubeApp/FreeTube#1380]
I've started using the SponsorBlock feature for a while and it has been working quite well as it should. However, I found the fact that you can't choose to see an ad without having to disable the feature in the settings quite bothersome.

My suggestion to tackle this issue would be to make the auto-skip an option in the SponsorBlock's settings.  Thus, when chosen, the program would function that same as currently does and when not, the user would manually skips the ad via a "skip this ad" button (in similar fashion of YouTube's ads).

I believe this to be the best option since it covers for the users that don't care the slightest for ads and those that sometimes do.

**Comment by GilgusMaximus** [comment]
A good idea, but as we are not actively supporting SponsorBlock as a team (see [disclaimer](https://docs.freetubeapp.io/usage/sponsorblock/)) , we have to ask someone from the community to implement this.

**Comment by ScriptForAll** [comment]
Okay, I understand. However, I have some comments regarding it:

Since the FreeTube team encourages content creators to use sponsors inside their videos, as it's stated in the aforementioned disclaimer, I reckon that the feature I've requested complies with said position since it's a better implementation that allows to skip ad segments only and only if the user wishes to do rather than doing it automatically. 
That being said, I believe that the team itself could support it if the decision is based on the idea of supporting content creators.

Hope this serves as food for thought and thanks for reading.

**Comment by GilgusMaximus** [comment]
I agree with you. But still, we do not want to work on Sponsorblock. We only merged this because @starsep put in the effort to make it. 
So he might be able to help or provide some guidance if needed.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
**Is your feature request related to a usage problem (not a bug)? Please describe.**
At the moment u always get forced to skip the sponsor. It would be nice that it doesn't force u to skip but still shows u the color of the sponsorblock segment. So the user has more control over what sponser he/she like to see or not.

**Describe the solution you'd like to see implemented**
Maybe this could done with a toggle in the settings?

**Describe alternatives you've considered**
This comment was made by another user but i cant ping him/her :( 

The way I've seen it done in most other implementations is that for each category, each type of section (like sponsor or self promotion) you have several settings. You can disable it completely, you can show the bar, you can show the bar and skip, and in some implementations you can show the bar and a button to skip manually. Probably the first 3 are what the vast majority of people use.

So you could say "I want to show the sponsor section and the self promotion section, skip only the sponsor section, and I don't care about all the other sections".

For now though, having an automatic skip of all sponsor sections is I think what the majority of people want and need from a sponsorblock implementation. I've been using the fork as my daily FreeTube experience and it's been great. I'll change it to the development branch of this repo to see if I get any issues and continue testing it.

**Comment by rpc31** [comment]
The ability to see the colored segments without the auto-skip enabled feature would be great, the same way the extension can work with youtube.com

**Comment by EvilGremlin** [comment]
+1, SponsorBlock should be set on per-channel basis with toggles in player for channel (retained) and just now playing video (with hotkey)

**Comment by bagajohny** [comment]
I would also like to have similar functionality to original sponsorblock of browser extension where you get a option to manually skip sponsors/segments with Enter key

**Comment by cra3yk** [comment]
Any chance to add this feature? (option to manually skip sponsors/segments with Enter key)

**Comment by KaKi87** [comment]
Hello @absidue, any news on this please ? Thanks :pray:

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Sponsorblock does the following by default. Skip segment and if user would like to unskip they can press the enter key, same goes for reskip.

https://github.com/user-attachments/assets/b9b4467a-c6ef-44dd-bf91-96ee64fd4915
